src/context/WalletContext.tsx

```diff
@@ -45,15 +47,46 @@ const defaultState: WalletContextInterface = {
 
 const WalletContext = createContext<WalletContextInterface>(defaultState);
 
+async function getChainId(web3: Web3): Promise<`0x${string}`> {
+  const chainId = await web3.eth.getChainId();
+  return `0x${chainId.toString(16)}`;
+}
+
 export function WalletContextProvider({ children }: PropsWithChildren): JSX.Element {
   const [web3, setWeb3] = useState<Web3 | null>(null);
+  const [chainId, setChainId] = useState('');
+
+  console.log(chainId);
+
+  useEffect(() => {
+    if (web3 === null || !web3.currentProvider) return;
+    const provider = web3.currentProvider;
+
+    // there is typing error https://github.com/web3/web3.js/issues/5939
+    // @ts-ignore
+    provider.on('chainChanged', (chainId: unknown) => {
+      if (typeof chainId !== 'string') return;
+      // TODO: validate if is valid chain ID
```

Comment:
why is this todo, just hardcode supported chains (sepolia, moonbeam, mumbai)^^

Reply:
assume that some other wallet can give hex result without `0x` prefix